### PR TITLE
YAML validation with Yamale lib

### PR DESCRIPTION
**Pull request by mserranom**
This is a proof of concept of validation with Yamale library. It's not intented to merge, but potentially we can migrate all validations following this branch if we fancy.

I think it's a very powerful tool, this approach is quite clean and less verbose, and supports reusing easily custom validations.

Check my inline comments and please provide some feedback

### CODE REVIEW

File: tests/runner/local_runner_test.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 
 def test_local_runner():
-    local_runner = LocalRunner(['tests/runner/data/raw.json'], '', 'tests/runner/data/sample.yml')
```
Comment (mserranom) [inline tests/runner/local_runner_test.py]: this is just a yapf formatting change

File: blurr/core/base.py [hunk 2]
```diff
@@ -17,12 +35,24 @@ class BaseSchema(ABC):
     ATTRIBUTE_TYPE = 'Type'
     ATTRIBUTE_WHEN = 'When'
 
+    SCHEMA = '''
+        Name: str(min=1)
+        Type: dtc_type()
```
Comment (indradeep) [inline blurr/core/base.py]: Type t different levels have different meanings.  The typeloader that loads the type into the system will validate if the type name is valid.  

I'm not sure that we want to define the DTC type in the base.

File: blurr/core/base.py [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
-from blurr.core.loader import TypeLoader
+from blurr.core.loader import TypeLoader, SCHEMA_MAP
+
+
+# This would be in a more central place along with other validation code
+class DTCType(Validator):
+    TAG = 'dtc_type'
+
+    # all valid types would go here, I just made this up
+    VALUES = list(SCHEMA_MAP.keys()) + ['ProductML:DTC:Streaming']
```
Comment (indradeep) [inline blurr/core/base.py]: This layer is not aware of DTC / Streaming.

**Comment by mserranom** [comment]
closing, I'll raise a separate PR with the production-quality code

**Pull request closed without merging**